org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/configuration/BuildConfigurationBuilder.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core.configuration;
+
+/**
+ * Creates {@link BuildConfiguration} instances.
+ *
+ * @author Donat Csikos
+ * @since 3.0
+ */
+public interface BuildConfigurationBuilder {
```

Comment:
❌ This doesn't need to be an interface, you can just make the Builder a nested class of BuildConfiguration and give it a private constructor.

Reply:
I see, but I'd prefer to hide all implementation details in a separate class.

Reply:
For what purpose? What is there to hide? Let's not make it unnecessarily complex :)

Reply:
By that explanation `java.lang.Integer` should be an interface and there should be `IntegerImpl` and a factory method. But ask yourself: Did you *ever* have any problem with the fact that Integer has all the code it needs inside it?

Reply:
Sure, I understand that. I just have the implementation already and I quite liked it the way it is. I'll change that soon.

Reply:
> I'll soon adjust the implemementation.

I.e. you'll remove the unnecessary indirection?

Reply:
Yes.

Reply:
Fixed.